pkg/await/awaiters.go

```diff
@@ -406,6 +406,86 @@ func untilCoreV1ServiceInitialized(
 
 // --------------------------------------------------------------------------
 
+// core/v1/ServiceAccount
+
+// --------------------------------------------------------------------------
+
+func untilCoreV1ServiceAccountInitialized(
+	clientForResource dynamic.ResourceInterface, obj *unstructured.Unstructured,
+) error {
+	//
+	// A ServiceAccount is considered initialized when the controller adds the default secret to the
+	// secrets array (i.e., in addition to the secrets specified by the user).
+	//
+
+	specSecrets, _ := pluck(obj.Object, "secrets")
+	var numSpecSecrets int
+	if specSecretsArr, isArr := specSecrets.([]interface{}); isArr {
```

Comment:
What could this be otherwise, if not an array?

Reply:
It can be nothing other than an array. If we get something other than an array, that's a bug in client-go.